jetpack.php

```diff
@@ -133,6 +133,9 @@ function jetpack_admin_unsupported_php_notice() { ?>
 	return;
 }
 
+// Load all the packages.
+require plugin_dir_path( __FILE__ ) . '/vendor/autoload.php';
```

Comment:
Do you think we could add a safeguard here, to avoid fataling sites where `composer install` hasn't been run yet? Something like:

```
diff --git a/jetpack.php b/jetpack.php
index 0f450a7d9..59bd48868 100644
--- a/jetpack.php
+++ b/jetpack.php
@@ -118,6 +118,27 @@ function jetpack_admin_unsupported_php_notice() { ?>
 	<?php
 }
 
+/**
+ * Outputs an admin notice for folks running Jetpack without having run composer install.
+ *
+ * @since 7.4.0
+ */
+function jetpack_admin_missing_autoloader() { ?>
+	<div class="notice notice-error is-dismissible">
+		<p>
+		<?php
+		printf(
+			/* translators: Placeholder is a link to a support document. */
+			__( 'Your installation of Jetpack is incomplete. If you installed Jetpack from GitHub, please refer to <a href="%1$s" target="_blank" rel="noopener noreferrer">this document</a> to set up your development environment.', 'jetpack' ),
+			esc_url( 'https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md' )
+		);
+		?>
+		</p>
+	</p>
+	</div>
+	<?php
+}
+
 if ( version_compare( phpversion(), JETPACK__MINIMUM_PHP_VERSION, '<' ) ) {
 	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
 		error_log(
@@ -134,7 +155,23 @@ if ( version_compare( phpversion(), JETPACK__MINIMUM_PHP_VERSION, '<' ) ) {
 }
 
 // Load all the packages.
-require plugin_dir_path( __FILE__ ) . '/vendor/autoload.php';
+$jetpack_autoloader = JETPACK__PLUGIN_DIR . '/vendor/autoload.php';
+if ( is_readable( $jetpack_autoloader ) ) {
+	require $jetpack_autoloader;
+} else {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log(
+			sprintf(
+				/* translators: Placeholder is a link to a support document. */
+				__( 'Your installation of Jetpack is incomplete. If you installed Jetpack from GitHub, please refer to <a href="%1$s" target="_blank" rel="noopener noreferrer">this document</a> to set up your development environment.', 'jetpack' ),
+				esc_url( 'https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md' )
+			)
+		);
+	}
+	add_action( 'admin_notices', 'jetpack_admin_missing_autoloader' );
+	return;
+}
+
 
 add_filter( 'jetpack_require_lib_dir', 'jetpack_require_lib_dir' );
 add_filter( 'jetpack_should_use_minified_assets', 'jetpack_should_use_minified_assets', 9 );

```

Reply:
🚢 IT

Reply:
done in 500e6645a5a3fd22b526877fc086cfc30addea66